cpp/src/copying/segmented_shift.cu

```diff
@@ -121,6 +121,14 @@ struct segmented_shift_functor {
   template <typename... Args>
   std::unique_ptr<column> operator()(Args&&...)
   {
+    if constexpr (std::is_same_v<T, cudf::list_view>) {
+      CUDF_FAIL("segmented_shift does not support list_view yet");
+    }
+
+    if constexpr (std::is_same_v<T, cudf::struct_view>) {
+      CUDF_FAIL("segmented_shift does not support struct_view yet");
+    }
```

Comment:
Based on reviewing #8239 I'm confident we do not need all three of these messages and only the final generic message should be sufficient.